packages/form-layout/src/vaadin-form-item.js

```diff
@@ -159,17 +160,62 @@ class FormItem extends ThemableMixin(PolymerElement) {
     return 'vaadin-form-item';
   }
 
+  constructor() {
+    super();
+    this.__updateInvalidState = this.__updateInvalidState.bind(this);
+    this.__contentFieldObserver = new MutationObserver(() => this.__updateRequiredState(this.__contentField.required));
+  }
+
   /** @private */
   _onLabelClick() {
-    const firstContentElementChild = Array.prototype.find.call(
-      this.$.contentSlot.assignedNodes(),
-      (e) => e.nodeType === Node.ELEMENT_NODE
-    );
+    const firstContentElementChild = this.__getContentChildNodes()[0];
     if (firstContentElementChild) {
       firstContentElementChild.focus();
       firstContentElementChild.click();
     }
   }
+
+  /** @private */
+  __getContentChildNodes() {
+    return [...this.$.contentSlot.assignedNodes()].filter((e) => e.nodeType === Node.ELEMENT_NODE);
```

Comment:
Looks like we can use [`assignedElements()`](https://caniuse.com/mdn-api_htmlslotelement_assignedelements) instead, it's available since Safari 12.1.

```suggestion
    return this.$.contentSlot.assignedElements();
```